### PR TITLE
Improve CSV File Handling

### PR DESCRIPTION
**Pull request by CakeVR**
This PR aims to improves handling CSV file creation and improve code quality.

- Adds class for CSV file handling, includes related CSV features.
- Adds `Save file mode` allowing to save CSVs next to timelines or inside the `Translation file path`.
- Fixes `Update CSV files` not finding all timelines outside of `Translation file path`.
- Adds `Save file mode` setting to signal receivers saving settings.
- Disables `Update CSV files` if `Translation file path` is required and invalid.
- Updates related tooltips.

Tasks left to do:
- [x] Disable `Update CSV files` button, if `Translation file path` is empty and a project CSV file is used or the save file mode requires it.
- [x] Further testing.

### CODE REVIEW

File: addons/dialogic/Editor/Settings/settings_translation.gd [hunk 1]
```diff
@@ -29,33 +31,75 @@ func _ready() -> void:
...
 	ProjectSettings.set_setting('dialogic/translation/translation_folder', %TransFolderPicker.current_value)
 	ProjectSettings.set_setting('internationalization/locale/test', %TestingLocale.current_value)
+	ProjectSettings.set_setting('internationalization/save_mode', %SaveLocationMode.selected)
 	ProjectSettings.save()
 
+## Checks whether the translation folder path is required.
+## If it is, disables the "Update CSV files" button and shows a warning.
+##
+## The translation folder path is required when either of the following is true:
+## - The translation mode is set to "Per Timeline".
```
Comment (Jowan-Spooner) [inline addons/dialogic/Editor/Settings/settings_translation.gd:77]: Is this what you actually meant. Intuitively it seems that it should be needed when translation mode is "Per Project" or if it's ("Per Timeline" and save_location == "Inside Translation Folder")

File: addons/dialogic/Editor/Settings/csv_file.gd [hunk 2]
```diff
@@ -0,0 +1,122 @@
+extends RefCounted
```
Comment (Jowan-Spooner) [inline addons/dialogic/Editor/Settings/csv_file.gd:1]: It should be class_name xxx then extends on the next line.

File: addons/dialogic/Editor/Settings/csv_file.gd [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+## The amount of events that were added to the CSV file.
+var new_events: int = 0
+
+## Attempts to load the CSV file from [param file_path].
+## If the file does not exist, a single entry is added to the [member lines]
+## array.
+func _init(file_path: String, original_locale: String) -> void:
+    used_file_path = file_path
+    print("Opening CSV file from path: ", file_path)
```
Comment (Jowan-Spooner) [inline addons/dialogic/Editor/Settings/csv_file.gd:35]: Can we get rid of the print?

File: addons/dialogic/Editor/Settings/csv_file.gd [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+    file = FileAccess.open(file_path, FileAccess.READ)
+
+    var locale_csv_row := file.get_csv_line()
+    column_count = locale_csv_row.size()
+    var locale_key := locale_csv_row[0]
+
+    old_lines[locale_key] = locale_csv_row
+
+    _read_file_into_lines()
+
```
Comment (Jowan-Spooner) [inline addons/dialogic/Editor/Settings/csv_file.gd:61]: Separate methods by two empty lines please.

**Review by Jowan-Spooner — APPROVED** [review]
Looks a lot better. Left some small style comments. Haven't battle-tested this but seems to work well on a first glance.

File: addons/dialogic/Editor/Settings/settings_translation.gd (hunk 1 above)
Comment (CakeVR) [reply addons/dialogic/Editor/Settings/settings_translation.gd]: I wanted to write "Per Project" , fixed, thanks.

**Comment by Jowan-Spooner** [comment]
Lgtm. Thanks!

**Pull request merged**